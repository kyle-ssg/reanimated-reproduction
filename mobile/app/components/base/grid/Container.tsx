import React from 'react'

export type ContainerType = {
  children?: React.ReactNode
  ml0?: boolean
  ml5?: boolean
  ml10?: boolean
  ml15?: boolean
  ml20?: boolean
  mr0?: boolean
  mr5?: boolean
  mr10?: boolean
  mr15?: boolean
  mr20?: boolean
  style?: ReactNative.StyleProp<ReactNative.ViewStyle>
}

const Container: React.FC<ContainerType> = ({
  ml0,
  ml5,
  ml10 = true,
  ml15,
  ml20,
  mr0,
  mr5,
  mr10 = true,
  mr15,
  mr20,
  style,
  children,
}) => (
  <View
    style={[
      ml10 && Styles.ml10,
      mr10 && Styles.mr10,
      ml0 && Styles.ml0,
      ml5 && Styles.ml5,
      ml15 && Styles.ml15,
      ml20 && Styles.ml20,
      mr0 && Styles.mr0,
      mr5 && Styles.mr5,
      mr15 && Styles.mr15,
      mr20 && Styles.mr20,
      style,
    ]}
  >
    {children}
  </View>
)

export default Container
