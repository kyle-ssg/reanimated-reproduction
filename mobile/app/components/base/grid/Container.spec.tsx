import * as React from "react";
import Container from "./Container";
import renderer from "react-test-renderer";


describe("<Container />", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Container />);
    expect(tree).toMatchSnapshot();
  });
});
