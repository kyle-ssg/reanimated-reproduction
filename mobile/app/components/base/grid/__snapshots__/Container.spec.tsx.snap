// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 16,
      },
      Object {
        "marginRight": 16,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Flex /> renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
